master/buildbot/reporters/stash.py

```diff
@@ -61,8 +67,22 @@ def send(self, build):
         else:
             status = STASH_INPROGRESS
             description = self.startDescription
+
+        # got_revision could be a string, a dictionary or None
+        got_revision = props.getProperty('got_revision', None)
         for sourcestamp in build['buildset']['sourcestamps']:
             sha = sourcestamp['revision']
+
+            if sha is None:
+                if isinstance(got_revision, dict):
```

Comment:
ah I did not see that. if the revision is available in the event we should set it.

If there is a new push, then we will get a new event, which in turn will trigger a new build testing that new sha.
The new event may cancel the previous build, this feature is not implemented, but for me, this is racy to expect the previous build to take latest version.

For all the github style pull request tester, (travis, appveyor) this is done like this, so I think we should stay coherent with this behaviour.